blottertrax/helper/title_parser.py

```diff
@@ -1,53 +1,140 @@
-import re
-
-from blottertrax.value_objects.parsed_submission import ParsedSubmission
-
-
-class TitleParser:
-
-    @staticmethod
-    def create_parsed_submission_from_submission(submission) -> ParsedSubmission:
-        artist = None
-        feature_artist = None
-        song_title = None
-        post_title = submission.title
-
-        for dash in [' - ', ' — ', ' -- ', ' - - ', ' —— ']:
-            if dash in post_title:
-                split_result = post_title.split(dash)
-                artist = split_result[0]
-                song_title = split_result[1]
-
-                break
-
-        for feature in [' feat. ', ' ft. ', ' feature ', ' featuring ', ' & ']:
-            if feature not in post_title:
-                continue
-
-            feat_index = post_title.index(feature)
-
-            # remove featuring from artist if exists
-            if len(artist) > feat_index:
-                artist = artist[:feat_index].strip()
-
-            feat_index += len(feature)
-
-            # isolate featuring artist
-            feature_artist = post_title[feat_index:].strip()
-
-            for end_char in [' -', ')', '[', ' —']:
-                if end_char in feature_artist:
-                    split_result = feature_artist.split(end_char)
-                    feature_artist, song_title = split_result
-
-                    break
-
-        # Remove everything between () and []
-        song_title = re.sub(r"\(.*\)", "", song_title)
-        song_title = re.sub(r"[.*]", "", song_title)
-        song_title = song_title.strip()
-
-        if artist is not None:
-            return ParsedSubmission(artist, feature_artist, song_title, submission.url)
-
-        raise LookupError
+from blottertrax.value_objects.parsed_submission import ParsedSubmission
+
+class TitleParser:
+
+    # Calls the main parsing methods and returns a parsed submission if successful.
+    @classmethod
+    def create_parsed_submission_from_submission(self, submission) -> ParsedSubmission:
+        artist = self.get_artist_name_from_submission_title(submission.title)
+        if artist[0] is not None:
```

Comment:
Not sure we actually need this check as it was already checked.

`  if return_list[0] is not None:
     return return_list
`
https://github.com/martijnboers/BlotterTrax/pull/48/files#diff-fe0076842fd5a3dcf6a1829ff40423a3R89

Reply:
Good catch